processing/src/main/java/org/apache/druid/query/groupby/orderby/DefaultLimitSpec.java

```diff
@@ -220,6 +221,15 @@ public boolean isLimited()
       sortingNeeded = !query.getGranularity().equals(Granularities.ALL) && query.getContextSortByDimsFirst();
     }
 
+    if (!sortingNeeded) {
+      Map<String, Object> timestampFieldContext = GroupByQueryHelper.findTimestampResultField(query);
+      if (!timestampFieldContext.isEmpty()) {
+        int timestampResultFieldIndex = (int) timestampFieldContext.get(GroupByQuery.CTX_TIMESTAMP_RESULT_FIELD_INDEX);
+        // change the sorting order when timestampResultField is neither first nor last dimension
+        sortingNeeded = timestampResultFieldIndex != 0 && timestampResultFieldIndex != query.getDimensions().size() - 1;
```

Comment:
Should this depend on `query.getContextSortByDimsFirst()`? Such as 

```
        sortingNeeded = query.getContextSortByDimsFirst()
                        ? timestampResultFieldIndex != query.getDimensions().size() - 1
                        : timestampResultFieldIndex != 0;
```

Reply:
yes, your code is more easier understanding. I will fix it.